qiita_pet/templates/study_base.html

```diff
@@ -121,6 +142,22 @@
     }
   }
 
+  function set_admin_visibility(visibility, aid) {
+    $.post('/admin/artifact/', {artifact_id: +aid, visibility: visibility})
+      .done(function ( data ) {
+        if(data["status"] == "error") {
+          bootstrapAlert(data.message.replace("\n", "<br/>"), "danger", 4000);
```

Comment:
Code duplication and magic number 4000. Explain number and put "danger"/"info" in a variable and use a single call to bootstrapAlert.